dotCMS/src/main/java/com/dotmarketing/business/VersionableFactoryImpl.java

```diff
@@ -381,15 +381,29 @@ protected Optional<ContentletVersionInfo> findContentletVersionInfoInDB(String i
 	@Override
 	public Optional<ContentletVersionInfo> findAnyContentletVersionInfo(final String identifier)
 			throws DotDataException {
-		final List<ContentletVersionInfo> result = findContentletVersionInfos(identifier, 1);
+		final List<ContentletVersionInfo> result = findContentletVersionInfos(identifier, null, 1);
 		return !result.isEmpty() ? Optional.of(result.get(0)) : Optional.empty();
 	}
 
 	private List<ContentletVersionInfo> findContentletVersionInfos(final String identifier,
 			final int maxResults) throws DotDataException, DotStateException {
-		final DotConnect dotConnect = new DotConnect()
-				.setSQL("SELECT * FROM contentlet_version_info WHERE identifier=?")
-				.addParam(identifier);
+		return 	findContentletVersionInfos(identifier, null, maxResults);
+	}
+
+	private List<ContentletVersionInfo> findContentletVersionInfos(final String identifier, final String variantName,
+			final int maxResults) throws DotDataException, DotStateException {
+		final DotConnect dotConnect = new DotConnect();
+
+		if (UtilMethods.isSet(variantName)) {
+			dotConnect.setSQL(
+							"SELECT * FROM contentlet_version_info WHERE identifier=? AND variant_id = ?")
+					.addParam(identifier)
+					.addParam(variantName);
```

Comment:
it would be great to have something such as
.addParamIfSet(variantName) so you can reduce it to
```
dotConnect.setSQL(UtilMethods.isSet(variantName)?
							"SELECT * FROM contentlet_version_info WHERE identifier=? AND variant_id = ?":"SELECT * FROM contentlet_version_info WHERE identifier=?")
					.addParam(identifier)
					.addParamIfSet(variantName);
					```

Reply:
I like that idea but I think is better send it in another PR